traveler-swift-core/TravelerKit/Models/BookingItemSearchParameters.swift

```diff
@@ -64,16 +64,18 @@ struct BookingItemSearchParameters: Decodable {
         } else {
             self.priceRange = nil
         }
-
-        if let startDate = try container.decode(Date?.self, forKey: .startDate),
-            let endDate = try container.decode(Date?.self, forKey: .endDate) {
+        
+        if let startDateString = try? container.decode(String.self, forKey: .startDate),
+            let startDate = ISO8601DateFormatter.dateOnlyFormatter.date(from: startDateString),
+            let endDateString = try? container.decode(String.self, forKey: .endDate),
+            let endDate = ISO8601DateFormatter.dateOnlyFormatter.date(from: endDateString) {
```

Comment:
What is the need of all this?. From backend you receive a full iso date, and we expect to parse a complete date (with no formatting). If formatting is required that should be done at the implementation layer. We can't expect that people using the SDK will only want date only all the time.

Reply:
When I leave it as it was
`if let startDate = try container.decode(Date?.self, forKey: .startDate),
            let endDate = try container.decode(Date?.self, forKey: .endDate) {
            let range = startDate...endDate

            self.dateRange = DateRangeFilter(range: range)
        } else {
            self.dateRange = nil
        }`

It always fails here. Perhaps dates sent from the backend are in the wrong format.
This is what we get from the backend:
```
availabilityStart : "2020-03-18T14:57:40"
availabilityEnd : "2020-08-31T00:00:00"
```
You can try running the app on the dev server. 
I don't get all results, because parsing always fails at date conversion for some reason.
I only get the first two groups of the response because other groups have searchParams and availabilityStart and availabilityEnd date (which fails to parse).

Reply:
Since other model classes uses the approach to parse date string and then convert string to date, we will use the same approach here.
But it all should be refactored in the future to parse iso8601 formatted string returned from the backend directly to Date.